packages/metal-component/test/Component.js

```diff
@@ -142,6 +165,37 @@ describe('Component', function() {
 			assert.strictEqual(comp, comp.attach());
 		});
 
+		it('should run "willDetach" lifecycle method when the component about to detach', function() {
+			class TestComponent extends Component {
+			}
+			sinon.spy(TestComponent.prototype, 'willDetach');
+			comp = new TestComponent();
+
+			assert.strictEqual(0, comp.willDetach.callCount);
+
+			comp.detach();
+
+			assert.ok(comp.willDetach.calledBefore(Component.prototype.detached));
+			assert.strictEqual(1, comp.willDetach.callCount);
+		});
+
+		it('should emit "willDetach" lifecycle method when the component about to detach', function() {
```

Comment:
Are we saying emit ... method in the rest of tests? maybe better to say emit ... lifecycle **event** when the component **is** about to...?